include/common.h

```diff
@@ -58,3 +59,82 @@ inline double wrap_pi(double x) {
 
   return x;
 }
+
+inline bool CheckConfigElement(const TiXmlElement &config, std::string param) {
+  const TiXmlElement *e = config.FirstChildElement(param);
+  return e != nullptr;
+}
+
+inline bool CheckConfigElement(const TiXmlHandle &config, std::string group, std::string param) {
+  const TiXmlElement *group_element = config.FirstChild(group).Element();
+  if (!group_element) {
+    return false;
+  }
+
+  const TiXmlElement *e = group_element->FirstChildElement(param);
+  return e != nullptr;
+}
+
+inline TiXmlElement GetXmlElement(const TiXmlHandle &config, std::string group, std::string param) {
+  const TiXmlElement *group_element = config.FirstChild(group).Element();
+  if (!group_element) {
+    return nullptr;
+  }
+
+  const TiXmlElement *e = group_element->FirstChildElement(param);
+  if (!e) {
+    return nullptr;
+  }
+  return *e;
+}
+
+template <typename T>
+void GetConfigElement(const TiXmlHandle &config, const std::string &group, const std::string &param, T &param_value) {
```

Comment:
Return a bool here and for the others below, so a caller can know whether it was filled out or not.